File: snippets/fsharp/System/TimeSpan/MaxValue/fields.fs
//<Snippet1>
// Example of the TimeSpan fields.
open System

// Pad the end of a TimeSpan string with spaces if it does not 
// contain milliseconds.
let align (interval: TimeSpan) =
    let intervalStr = string interval
    let pointIndex = intervalStr.IndexOf ':'

    let pointIndex = intervalStr.IndexOf('.', pointIndex)
    if pointIndex < 0 then intervalStr + "        "
    else intervalStr


printfn "This example of the fields of the TimeSpan class\ngenerates the following output.\n"
printfn "%-22s%18s" "Field" "Value"
printfn "%-22s%18s" "-----" "-----"

// Display the maximum, minimum, and zero TimeSpan values.
printfn $"""{"Maximum TimeSpan",-22}{align TimeSpan.MaxValue,26}"""
printfn $"""{"Minimum TimeSpan",-22}{align TimeSpan.MinValue,26}"""
printfn $"""{"Zero TimeSpan",-22}{align TimeSpan.Zero,26}
"""

// Display the ticks-per-time-unit fields.
printfn $"""{"Ticks per day",-22}{TimeSpan.TicksPerDay,18:N0}"""
printfn $"""{"Ticks per hour",-22}{TimeSpan.TicksPerHour,18:N0}""" 
printfn $"""{"Ticks per minute",-22}{TimeSpan.TicksPerMinute,18:N0}""" 
printfn $"""{"Ticks per second",-22}{TimeSpan.TicksPerSecond,18:N0}""" 
printfn $"""{"Ticks per millisecond",-22}{TimeSpan.TicksPerMillisecond,18:N0}""" 

(*
This example of the fields of the TimeSpan class
generates the following output.

Field                              Value
-----                              -----
Maximum TimeSpan       10675199.02:48:05.4775807
Minimum TimeSpan      -10675199.02:48:05.4775808
Zero TimeSpan                   00:00:00

Ticks per day            864,000,000,000
Ticks per hour            36,000,000,000
Ticks per minute             600,000,000
Ticks per second              10,000,000
Ticks per millisecond             10,000
*)
//</Snippet1>

File: snippets/csharp/System/Decimal/MinusOne/fields.cs
//<Snippet1>
// Example of the Decimal fields.
using System;

class DecimalFieldsDemo
{
    public static void Main()
    {
        const string numberFmt = "{0,-25}{1,45:N0}";
        const string exprFmt = "{0,-55}{1,15}";

        Console.WriteLine(
            "This example of the fields of the Decimal structure " +
            "\ngenerates the following output.\n");
        Console.WriteLine(numberFmt, "Field or Expression", "Value");
        Console.WriteLine(numberFmt, "-------------------", "-----");

        // Display the values of the Decimal fields.
        Console.WriteLine(numberFmt, "Decimal.MaxValue",
            decimal.MaxValue);
        Console.WriteLine(numberFmt, "Decimal.MinValue",
            decimal.MinValue);
        Console.WriteLine(numberFmt, "Decimal.MinusOne",
            decimal.MinusOne);
        Console.WriteLine(numberFmt, "Decimal.One", decimal.One);
        Console.WriteLine(numberFmt, "Decimal.Zero", decimal.Zero);
        Console.WriteLine();

        // Display the values of expressions of the Decimal fields.
        Console.WriteLine(exprFmt,
            "( Decimal.MinusOne + Decimal.One ) == Decimal.Zero",
            (decimal.MinusOne + decimal.One) == decimal.Zero);
        Console.WriteLine(exprFmt,
            "Decimal.MaxValue + Decimal.MinValue",
            decimal.MaxValue + decimal.MinValue);
        Console.WriteLine(exprFmt,
            "Decimal.MinValue / Decimal.MaxValue",
            decimal.MinValue / decimal.MaxValue);
        Console.WriteLine($"{"100000000000000M / Decimal.MaxValue",-40}{100000000000000M / decimal.MaxValue,30}");
    }
}

/*
This example of the fields of the Decimal structure
generates the following output.

Field or Expression                                              Value
-------------------                                              -----
Decimal.MaxValue                79,228,162,514,264,337,593,543,950,335
Decimal.MinValue               -79,228,162,514,264,337,593,543,950,335
Decimal.MinusOne                                                    -1
Decimal.One                                                          1
Decimal.Zero                                                         0

( Decimal.MinusOne + Decimal.One ) == Decimal.Zero                True
Decimal.MaxValue + Decimal.MinValue                                  0
Decimal.MinValue / Decimal.MaxValue                                 -1
100000000000000M / Decimal.MaxValue     0.0000000000000012621774483536
*/
//</Snippet1>
